.buildkite/shared-pipeline-vars

```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+
+# This file is `source`'d before calling `buildkite-agent pipeline upload`, and can be used
+# to set up some variables that will be interpolated in the `.yml` pipeline before uploading it.
+
+# If you update the image to a newer Xcode version, don't forget to also update the badge in the README.md file accordingly for consistency
+export IMAGE_ID="xcode-15.4"
```

Comment:
@AliSoftware If I'm not mistaken, in the current configuration there are two places to set the Xcode image, so it is possible to set a beta Xcode for non-release builds.

While this creates a bit of code friction, it allows us to have betas using a beta Xcode and releases to be submitted to Apple using a stable Xcode version as they require.

Is the same possible here? Or does this affect all builds — including releases?

Reply:
Oh, I didn't know this was potentially intentional to have two different versions of Xcode for the different pipelines.

The only change I made in this PR is replacing the existing values with those constants. Before this PR the `IMAGE_ID` used in both `pipeline.yml` and `release-builds.yml` were already both `xcode-15.4` and the same, so that PR won't change that.

If you want to be able to change them at different pace, we could consider using two different constants — e.g. declare separate `BETA_IMAGE_ID` and `RELEASE_IMAGE_ID` values to be exported in this `shared-pipeline-vars` here, and then use `IMAGE_ID: $BETA_IMAGE_ID` on `pipeline.yml` to be used on PR builds, and `IMAGE_ID: $RELEASE_IMAGE_ID` on `release-builds.yml` to be used for beta and final release builds sent to TestFlight and App Store.

Reply:
In fact, we would like to have the final build use a stable Xcode version (so we can submit it for review) and all the rest to use the beta one (including TestFlight). This helps us in assessing issues because the TestFlight user base is quite active so we can fix them asap.

However, this is a bit confusing, happens just this time of the year and we can change the image manually ourselves if we want to. So I wonder if we should or should not reflect this on the code. 🤷

Reply:
In the current setup, we can have the VM images used to be different for `pipeline.yml` (the pipeline used for PR builds and builds made when new commits land in  `trunk`) vs `release-builds.yml` (the pipeline used to generate builds sent to TestFlight—both for beta or not)

But note that, even before this PR, this has always been the case that both betas (from `new_beta_release` lane) and final build (from `finalize_release` lane) use the same `release-builds.yml` pipeline, and thus always used the same Image and Xcode version on CI when those are built.
So I'm not sure how you were handling this even before the change from this PR 😅  because the Xcode version and VM image used by CI has always been the same for both beta and final builds, even if it could be different from the one used for building and running tests on PRs and `trunk`.

---

If you think that could be valuable, we can totally brainstorm how we could rewrite the pipelines to make it possible to use different VM images / Xcode versions for intermediate betas vs final builds, as opposed to only use the beta for PR builds. But that would be a new feature that wasn't supported even by the current setup before this PR 😉 

My initial thoughts about how to enable such a setup is to make use of the fact that [the `BETA_RELEASE` env var](https://github.com/Automattic/pocket-casts-ios/blob/trunk/fastlane/Fastfile#L971) is [set to `true` when the `release-builds.yml` pipeline is triggered for a beta](https://github.com/Automattic/pocket-casts-ios/blob/trunk/fastlane/Fastfile#L406), while it is [set to `false` when the same `release-builds.yml` pipeline is triggered for a final build](https://github.com/Automattic/pocket-casts-ios/blob/trunk/fastlane/Fastfile#L414).

So I guess something like this in the `shared_pipeline-vars` would cover it?
```bash
export IMAGE_ID=xcode-15.4
if [[ "$PIPELINE=="pipeline.yml" || "$BETA_RELEASE" == "true" ]];
  # Use the beta only for PRs, `trunk` and beta builds
  export IMAGE_ID=xcode-16.0-beta5
fi
```

Reply:
@leandroalonso I've updated https://github.com/Automattic/pocket-casts-ios/pull/2066 to have the logic mentioned in my above comment; let's continue the discussion about this approach in that other PR (since it's related to updating the image to the Xcode beta so feels a more relevant place to discuss strategies around it)